.github/workflows/event-processor.yml

```diff
@@ -30,12 +30,25 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: 'Sparse Checkout'
+        shell: pwsh
         run: |
-          set -ex
-          git clone --no-checkout --filter=tree:0 https://github.com/${{ github.repository }} .
+          $attempts = 0
+          while ($attempts -lt 3) {
```

Comment:
Should we consider hoisting "3" into a variable rather than hardcoding in a few areas so that its easier to tune the number of retries?